adept/container/actorlearner/learner_container.py

```diff
@@ -0,0 +1,374 @@
+            env.action_space,
+            net.internal_space(),
+            args.nb_env
+        )
+        actor = actor_cls.from_args(args, env.action_space)
+        learner = REGISTRY.lookup_learner(args.learner).from_args(args, rwd_norm)
+
+        exp_cls = REGISTRY.lookup_exp(args.exp).from_args(args, builder)
+
+        self.actor = actor
+        self.learner = learner
+        self.exp = exp_cls.from_args(args, builder).to(device)
+
+        # Rank 0 setup, load network/optimizer and create SummaryWriter/Saver
+        if rank == 0:
+            if args.load_network:
+                self.network = self.load_network(self.network, args.load_network)
+                logger.info('Reloaded network from {}'.format(args.load_network))
+            if args.load_optim:
+                self.optimizer = self.load_optim(self.optimizer, args.load_optim)
+                logger.info('Reloaded optimizer from {}'.format(args.load_optim))
+
+            logger.info('Network parameters: ' + str(self.count_parameters(net)))
+            self.summary_writer = SummaryWriter(log_id_dir)
+            self.saver = SimpleModelSaver(log_id_dir)
+
+    def run(self, workers, profile=False):
+        if profile:
+            try:
+                from pyinstrument import Profiler
+            except:
+                raise ImportError('You must install pyinstrument to use profiling.')
+            profiler = Profiler()
+            profiler.start()
+
+        # setup queuer
+        self.rollout_queuer = RolloutQueuerAsync(workers, self.nb_learn_batch, self.rollout_queue_size)
+        self.rollout_queuer.start()
+
+        # initial setup
+        global_step_count = self.initial_step_count
+        next_save = self.init_next_save(self.initial_step_count, self.epoch_len)
+        prev_step_t = time()
+        ep_rewards = torch.zeros(self.nb_env)
+        start_time = time()
+
+        # loop until total number steps
+        print('{} starting training'.format(self.rank))
+        while not self.done(global_step_count):
+            self.exp.clear()
+            # Get batch from queue
+            rollouts, terminal_rewards, terminal_infos = self.rollout_queuer.get()
+
+            # Iterate forward on batch
+            self.exp.write_exps(rollouts)
+            # keep a copy of terminals on the cpu it's faster
+            rollout_terminals = torch.stack(self.exp['terminals']).numpy()
+            self.exp.to(self.device)
+            r = self.exp.read()
+            internals = {k: ts[0].unbind(0) for k, ts in r.internals.items()}
+            for obs, rewards, terminals in zip(
+                    r.observations,
+                    r.rewards,
+                    rollout_terminals
+            ):
+                _, h_exp, internals = self.actor.act(self.network, obs,
+                                                     internals)
+                self.exp.write_actor(h_exp, no_env=True)
+
+                actual_terminals = np.where(terminals)[0]
```

Comment:
only the first terminal? i'm confused

Reply:
numpy where returns the indexes and values as a tuple, only the indexes are needed. updated naming and used unpacking syntax instead